lib/proxy/verify_runtime.rb

```diff
@@ -0,0 +1,52 @@
+module Proxy
+  class VerifyRuntime
+    class << self
+      # RFC5737 declares 192.0.2.0/24 as TEST-NET-1
+      MOCK_IP = '192.0.2.42'
+
+      def settings
+        Proxy::SETTINGS
+      end
+
+      def verify
+        {
+          reverse_proxy: verify_reverse_proxy,
+        }
+      end
+
+      def verify_reverse_proxy
+        # It's valid if there is no Foreman URL
+        return true unless settings.foreman_url
+
+        # Only needed for templates / registration
+        # TODO: make this more generic
+        return true unless ::Proxy::Plugins.instance.any? { |p| p[:state] == :running && ['templates', 'registration'].include?(p[:name]) }
```

Comment:
templates has a `template_url` setting while registration uses the global `foreman_url` setting.